src/util/Synchronized.h

```diff
@@ -33,7 +33,11 @@ struct AllowsSharedLocking<
                    decltype(std::declval<M&>().unlock_shared())>>
     : std::true_type {};
 
-/// A very simple spin lock, stolen from cppreference
+/// A very simple spin lock, stolen from cppreference. A spin lock is simply a
+/// lock that actively waits (as long as is necessary) for the lock to be
+/// released before locking it. In particular, this is OK when using it for
+/// serializing concurrent accesses to an object that are very simple and
+/// fast.00
```

Comment:
In particular, this is OK when using it for serializing simple and fast concurrent accesses to an object.